Modules/Broker/BrokerModule.lua

```diff
@@ -0,0 +1,214 @@
+--[[
+    Copyright (C) 2023 GurliGebis
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+]]
+
+local addonName, _ = ...
+local BattlePetCompletionist = LibStub("AceAddon-3.0"):GetAddon(addonName)
+local BrokerModule = BattlePetCompletionist:NewModule("BrokerModule", "AceEvent-3.0")
+local ConfigModule = BattlePetCompletionist:GetModule("ConfigModule")
+local DataModule = BattlePetCompletionist:GetModule("DataModule")
+local LibDataBroker = LibStub("LibDataBroker-1.1")
+local LibPetJournal = LibStub("LibPetJournal-2.0")
+
+-- Also used by MinimapModule
+function BrokerModule:GetDataObject()
+    return self.dataSource
+end
+
+function BrokerModule:RegisterEventHandlers()
+    self:RegisterEvent("ZONE_CHANGED_NEW_AREA", "RefreshData")
+    LibPetJournal.RegisterCallback(self, "PetListUpdated", "RefreshData")
+end
+
+function BrokerModule:GetZonePetData()
+    local mapId = C_Map.GetBestMapForUnit("player")
+    return DataModule:GetPetsInMap(mapId) or {}
+end
+
+function BrokerModule:OnInitialize()
+    self.dataSource = LibDataBroker:NewDataObject(addonName, {
+        type = "data source",
+        label = "BattlePets",
```

Comment:
Can you change it to `Battle Pets` instead? (With a space between the two words)

Reply:
Will do.

Reply:
Thanks

Reply:
Change applied.